src/models/branches/lines.jl

```diff
@@ -1,13 +1,41 @@
+function CheckThermalLimits(name::String,connectionpoints::@NT(from::Bus,to::Bus), r::Float64, x::Float64,rate::@NT(from_to::Float64, to_from::Float64),anglelimits::@NT(max::Float64, min::Float64))
+    theta_max = max(abs(anglelimits.min), abs(anglelimits.max))
+    r = r
+    x = x
+    g =  r / (r^2 + x^2)
+    b = -x / (r^2 + x^2)
+    y_mag = sqrt(g^2 + b^2)
+    fr_vmax = connectionpoints.from.voltagelimits.max
+    to_vmax =  connectionpoints.to.voltagelimits.max
+    m_vmax = max(fr_vmax, to_vmax)
+    c_max = sqrt(fr_vmax^2 + to_vmax^2 - 2*fr_vmax*to_vmax*cos(theta_max))
+    rating_from_to = Float64[]
+    rating_to_from  = Float64[]
+    new_rate = y_mag*m_vmax*c_max
+
+    rate.from_to <= 0.0 ? rating_from_to = new_rate : rate.from_to > new_rate ? rating_from_to = new_rate : rating_from_to = rate.from_to
+    rate.to_from <= 0.0 ? rating_to_from = new_rate : rate.to_from > new_rate ? rating_to_from = new_rate : rating_to_from = rate.to_from
+    
+    return @NT(from_to = rating_from_to, to_from = rating_to_from)
+end
+
 struct Line <: Branch
     name::String
     available::Bool
     connectionpoints::@NT(from::Bus, to::Bus)
     r::Float64 #[pu]
     x::Float64 #[pu]Co
     b::@NT(from::Float64, to::Float64) #[pu]
-    # TODO: add a rate and angle consistency check
-    rate::Float64 #[MVA]
+    # TODO: angle consistency check
+    rate::@NT(from_to::Float64, to_from::Float64)
     anglelimits::@NT(max::Float64, min::Float64)
+
+    function Line(name, available, connectionpoints, r, x, b, rate, anglelimits)
+
+        rating =  CheckThermalLimits(name, connectionpoints, r, x,  @NT(from_to = rate, to_from = rate), anglelimits)
```

Comment:
Rename to `calculate_thermal_limits`.

Reply:
@sourabhdalvi move this to the checks in utils and rename please